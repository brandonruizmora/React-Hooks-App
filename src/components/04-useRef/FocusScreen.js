import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = function () {
        inputRef.current.select();
        console.log(inputRef)
    }

    return (
        <div className="m-5">
            <h1>Focus Screen</h1>
            <hr />
            <input ref={ inputRef} className="form-control" />
            <button 
                className="btn btn-outline-primary mt-3" 
                onClick={ handleClick }
            >Focus</button>
        </div>
    )
}
