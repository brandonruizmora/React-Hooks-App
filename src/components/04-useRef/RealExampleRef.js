import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {
    const [show, setShow] = useState(false);
    return (
        <div className="m-5">
            <h1>RealExampleRef</h1>
            <hr />
            {show && <MultipleCustomHooks />}
            <button
                className="btn btn-primary"
                onClick={() => {
                    setShow(!show);
                }}
            >
                Show/Hide
            </button>
        </div>
    )
}
