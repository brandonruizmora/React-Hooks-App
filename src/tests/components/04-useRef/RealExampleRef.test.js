import '../../../setupTest';
import { shallow } from "enzyme";
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Pruebas en <RealExampleRef />', () => {

    test('Debe de mostrarse correctamente', () => {
        const wrapper = shallow(<RealExampleRef />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrarse el componente <MultipleCustomHooks />', () => {
        const wrapper = shallow(<RealExampleRef />);
        wrapper.find('button').simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
    
});
