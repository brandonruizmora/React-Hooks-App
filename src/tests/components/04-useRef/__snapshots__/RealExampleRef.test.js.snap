// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> Debe de mostrarse correctamente 1`] = `
<div
  className="m-5"
>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Pruebas en <RealExampleRef /> Debe de mostrarse el componente <MultipleCustomHooks /> 1`] = `
<div
  className="m-5"
>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
